lib/Net/Async/Blockchain/Omni.pm

```diff
@@ -134,28 +136,28 @@ async sub transform_transaction {
     return undef unless $received_transaction && $received_transaction->{ismine};
 
     my $amount = Math::BigFloat->new($received_transaction->{amount});
-    my $fee = Math::BigFloat->new($received_transaction->{fee} // 0);
-
+    my $fee    = Math::BigFloat->new($received_transaction->{fee} // 0);
+    my $block  = Math::BigInt->new($received_transaction->{block});
     my ($from, $to) =
         await Future->needs_all(map { $self->rpc_client->validate_address($received_transaction->{$_}) } qw(sendingaddress referenceaddress));
 
+    my %category;
+
+    for my $tx ($detailed_transaction->{details}->@*) {
+        $category{$tx->{category}} = 1;
+    }
+    my @categories = keys %category;
```

Comment:
just a tip you can use `uniq` from [List::Util](https://perldoc.perl.org/List/Util.html#uniq) to get the same result.

Reply:
Surely will keep in mind 👍